cmd/cstor-pool-mgmt/controller/backup-controller/new_backup_controller.go

```diff
@@ -0,0 +1,246 @@
+	bkplistop := metav1.ListOptions{
+		LabelSelector: bkplabel,
+	}
+	bkplist, err := clientset.OpenebsV1alpha1().BackupCStors(metav1.NamespaceAll).List(bkplistop)
+	if err != nil {
+		return
+	}
+
+	for _, bkp := range bkplist.Items {
+		switch bkp.Status {
+		case apis.BKPCStorStatusInProgress:
+			//Backup was in in-progress state
+			laststat := findLastBackupStat(clientset, bkp)
+			updateBackupStatus(clientset, bkp, laststat)
+		case apis.BKPCStorStatusDone:
+			continue
+		default:
+			//Set backup status as failed
+			updateBackupStatus(clientset, bkp, apis.BKPCStorStatusFailed)
+		}
+	}
+}
+
+// updateBackupStatus will update the backup status to given status
+func updateBackupStatus(clientset clientset.Interface, bkp apis.BackupCStor, status apis.BackupCStorStatus) {
+	bkp.Status = status
+
+	_, err := clientset.OpenebsV1alpha1().BackupCStors(bkp.Namespace).Update(&bkp)
+	if err != nil {
+		glog.Errorf("Failed to update backup(%s) status(%s)", status, bkp.Name)
+		return
+	}
+}
+
+// findLastBackupStat will find the status of backup from last-backup
+func findLastBackupStat(clientset clientset.Interface, bkp apis.BackupCStor) apis.BackupCStorStatus {
+	lastbkpname := bkp.Spec.BackupName + "-" + bkp.Spec.VolumeName
+	lastbkp, err := clientset.OpenebsV1alpha1().BackupCStorLasts(bkp.Namespace).Get(lastbkpname, v1.GetOptions{})
+	if err != nil {
+		// Unable to fetch the last backup, so we will return fail state
+		glog.Errorf("Failed to fetch last backup:%s error:%s", lastbkpname, err.Error())
+		return apis.BKPCStorStatusFailed
+	}
+
+	// let's check if snapname matches with current snapshot name
+	if bkp.Spec.SnapName == lastbkp.Spec.SnapName || bkp.Spec.SnapName == lastbkp.Spec.PrevSnapName {
+		return apis.BKPCStorStatusDone
+	}
+
+	// lastbackup snap/prevsnap doesn't match with bkp snapname
+	return apis.BKPCStorStatusFailed
+}
+
+// handleBKPAddEvent is to handle add operation of backup controller
+func (c *BackupController) handleBKPAddEvent(bkp *apis.BackupCStor, q *common.QueueLoad) {
+	q.Operation = common.QOpAdd
+	glog.Infof("BackupCStor event added: %v, %v", bkp.ObjectMeta.Name, string(bkp.ObjectMeta.UID))
+	c.recorder.Event(bkp, corev1.EventTypeNormal, string(common.SuccessSynced), string(common.MessageCreateSynced))
+	c.enqueueBackupCStor(bkp, *q)
+}
+
+// handleBKPUpdateEvent is to handle add operation of backup controller
+func (c *BackupController) handleBKPUpdateEvent(oldbkp, newbkp *apis.BackupCStor, q *common.QueueLoad) {
+	glog.Infof("Received Update for backup:%s", oldbkp.ObjectMeta.Name)
+
+	if newbkp.ResourceVersion == oldbkp.ResourceVersion {
+		return
+	}
+
+	if IsDestroyEvent(newbkp) {
+		q.Operation = common.QOpDestroy
+		glog.Infof("BackupCstor Destroy event : %v, %v", newbkp.ObjectMeta.Name, string(newbkp.ObjectMeta.UID))
+		c.recorder.Event(newbkp, corev1.EventTypeNormal, string(common.SuccessSynced), string(common.MessageDestroySynced))
```

Comment:
Proper error handling, Indentation should be followed throughout the PR.
Please refer to my earlier comments on the same.